src/index.js

```diff
@@ -1,29 +1,56 @@
 import puppeteer from 'puppeteer';
+import readline from 'readline';
 import db from './utils/db';
 import ticketManager from './utils/tickets';
 
 const main = async () => {
+  console.log(`Starting service... <q! to exit>`);
+  var should_break = false;
+
+  // create interface for reading user input
+  const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout,
+  });
+
   console.log('Launching browser...');
   const browser = await puppeteer.launch();
 
-  console.log('Finding new tickets...');
-  const tickets = await db.getNewTickets();
+  console.log(`Awaiting new tickets...`);
+  // tell program to quit service
+  rl.on(`line`, input => {
+    if (input == `q!`) {
+      should_break = true;
+    }
+  });
 
-  try {
-    // loop through and process tickets
-    for (let ticket of tickets) {
-      await ticketManager.processTicket(browser, ticket);
+  while (true) {
```

Comment:
Right now this is pegging the CPU and database, because this is an infinite loop. The event loop doesn't have any time to breathe.

<img width="353" alt="image" src="https://user-images.githubusercontent.com/1558223/69195737-248ff900-0af2-11ea-8745-91ba5ca8932c.png">


Instead, use [`setInterval`](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setInterval) to have a periodic check - say every second, or five seconds. This is much kinder to whatever is running this service.

Reply:
Oh yeah, this is bad, very bad. I’ll update it. Also looking into having the api make calls to an express server within the hc. I’m thinking that should be the long term solution there.